### PR TITLE
Propose HIPE: Rich Schema's W3C Verifable Credentials 

### PR DESCRIPTION
**Pull request by ashcherbakov**

### CODE REVIEW

File: text/0160-rich-schema-w3c-credentials/README.md [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+```
+
+Then the corresponding **W3C Credential** for CL signature scheme may look as follows: 
+```
+{
+  "@context": [
+    "did:sov:2f9F8ZmxuvDqRiqqY29x6dx9oU4qwFTkPbDpWtwGbdUsrCD",
+    "https://www.w3.org/2018/credentials/v1"
+  ]
+  "@id": "did:sov:zWFF7g7SetGfaUVCn8U9d62oDYrUJLuUtcy619",
```
Comment (brentzundel) [inline text/0160-rich-schema-w3c-credentials/README.md]:
```suggestion
  "id": "did:sov:zWFF7g7SetGfaUVCn8U9d62oDYrUJLuUtcy619",
```

File: text/0160-rich-schema-w3c-credentials/README.md [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+  "@type": ["VerifiableCredential", "did:sov:4e9F8ZmxuvDqRiqqY29x6dx9oU4qwFTkPbDpWtwGbdUsrCD"],
```
Comment (brentzundel) [inline text/0160-rich-schema-w3c-credentials/README.md]:
```suggestion
  "type": ["VerifiableCredential", "did:sov:4e9F8ZmxuvDqRiqqY29x6dx9oU4qwFTkPbDpWtwGbdUsrCD"],
```

File: text/0160-rich-schema-w3c-credentials/README.md [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+ and one used by the corresponding Rich Schema and Mapping objects.
+ - `@id` is a unique DID of the credential.
+ - `@type` is an array of two values: common for all W3C Credentils `VerifiableCredential` type 
+ (see `https://www.w3.org/2018/credentials/v1` context), and ID of a Rich Schema the credential is created against.
+ - `credentialSchema` points to the corresponding Credential Definition. 
+ `cdf` is the type of Credential Definition object among Rich Schema objects.
+ -  `issuer` has a DID of the issuer who is the author of the Credential Definition on the Ledger.
+ - `issuanceDate`is the date of the issuance.
+ - `credentialSubject` contains a list of (nested) attributes and the corresponding values. The set of attributes 
+ matches the one defined by the Mapping object.
```
Comment (brentzundel) [inline text/0160-rich-schema-w3c-credentials/README.md]:
```suggestion
 - `id` is a unique DID of the credential.
 - `type` is an array of two values: common for all W3C Credentials `VerifiableCredential` type 
 (see `https://www.w3.org/2018/credentials/v1` context), and ID of the rich rchema the credential is created against.
 - `credentialSchema` points to the corresponding credential definition. 
 `cdf` is the type of credential definition object among rich schema objects.
 -  `issuer` is the DID of the issuer who is the controller of the credential definition on the ledger.
 - `issuanceDate`is the date of the issuance. This is the date after which the claims in the credential 
 are considered valid by the issuer, which may be different from the date the credential was signed
 by the issuer or received by the holder.
 - `credentialSubject` contains a list of (nested) claims and the corresponding values. The set of claims 
 matches the ones defined by the mapping.
```

File: text/0160-rich-schema-w3c-credentials/README.md [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+ -  `proof` contains a ZKP signature (of `CL2` type in this example). The content is specific to the signature type 
+ and should not be parsed by applications. 
+
+### Indy Node API
+As Indy credentials are never stored on the Ledger, no changes in `indy-node` repository are required 
+besides a general Rich Schema support (see [Rich Schema Objects Common](https://github.com/hyperledger/indy-hipe/tree/master/text/0120-rich-schemas-common)).
```
Comment (brentzundel) [inline text/0160-rich-schema-w3c-credentials/README.md]:
```suggestion
besides general Rich Schema support (see [Rich Schema Objects Common](https://github.com/hyperledger/indy-hipe/tree/master/text/0120-rich-schemas-common)).
```

File: text/0160-rich-schema-w3c-credentials/README.md [hunk 5]
```diff
@@ -0,0 +1,402 @@
...
+
+### Indy VDR API
+As Indy credentials are never stored on the Ledger, no changes in `indy-vdr` repository are required 
+besides a general Rich Schema support (see [Rich Schema Objects Common](https://github.com/hyperledger/indy-hipe/tree/master/text/0120-rich-schemas-common)).
```
Comment (brentzundel) [inline text/0160-rich-schema-w3c-credentials/README.md]:
```suggestion
besides general Rich Schema support (see [Rich Schema Objects Common](https://github.com/hyperledger/indy-hipe/tree/master/text/0120-rich-schemas-common)).
```

File: text/0160-rich-schema-w3c-credentials/README.md [hunk 6]
```diff
@@ -0,0 +1,402 @@
...
+
+### Indy Credx API
+
+The W3C and Rich Schema compatible protocol will be implemented in `indy-credx` under the following assumptions:
```
Comment (brentzundel) [inline text/0160-rich-schema-w3c-credentials/README.md]:
```suggestion
The W3C and Rich Schema compatible protocols will be implemented in `indy-credx` under the following assumptions:
```

File: text/0160-rich-schema-w3c-credentials/README.md [hunk 7]
```diff
@@ -0,0 +1,402 @@
...
+  but this is out-of-scope for the current HIPE.  
+
+|               | W3C compatible credentials | Non-W3C compatible credentials |
+| ------------- | ------------- | --- |
+| **Rich Schema**  | Yes  | No  |
+| **Old Schema**  | No  | Yes |
+  
+#### Relationship with Anoncreds version   
+Both W3C compatible and non-W3C compatible issuance and presentation protocols can work with either Anoncreds 1.0 (`CL`)
+math or Anoncreds 2.0 (`CL2`) math, so that Anoncreds math version is orthogonal to the Schema and credentials format.
```
Comment (brentzundel) [inline text/0160-rich-schema-w3c-credentials/README.md]:
```suggestion
or Anoncreds 2.0 (`CL2`); the Anoncreds version is orthogonal to the Schema and credentials format.
```

File: text/0160-rich-schema-w3c-credentials/README.md [hunk 8]
```diff
@@ -0,0 +1,402 @@
...
+Any Indy credential compatible with W3C standard MUST have the following properties:
+
+#### @context 
+JSON-LD Context. The value MUST be an ordered set where 
+ - the first item MUST be a URI with the value `https://www.w3.org/2018/credentials/v1`.
+ - other items MAY be `@id` of contexts used by the corresponding Rich Schema and Mapping 
+ (see [Rich Schema Context](https://github.com/hyperledger/indy-hipe/tree/master/text/0149-rich-schema-schema#context)).
+
+
+#### @id
```
[the remark this answers is not in the document]
Comment (ashcherbakov) [reply text/0160-rich-schema-w3c-credentials/README.md]: Initially I read W3C cred spec that `id` property is a must-have one, but now I see that it's optional, so I will remove it from the must-have list.
No, this `id` property is not related to revocation at all. I've added it just for compatilbity with W3C Credentials format (assuming it must be there).

**Pull request merged**